x/evm/handler_test.go

```diff
@@ -525,3 +525,79 @@ func (suite *EvmTestSuite) TestErrorWhenDeployContract() {
 
 	// TODO: snapshot checking
 }
+
+func (suite *EvmTestSuite) deployTestRevertContract() common.Address {
+	k := suite.app.EvmKeeper
+	nonce := k.GetNonce(suite.from)
+	msg := ethtypes.NewMessage(
+		suite.from,
+		nil,
+		nonce,
+		big.NewInt(0),
+		2000000,
+		big.NewInt(1),
+		nil,
+		nil,
+		types.TestRevert.Bin,
+		nil,
+		true,
+	)
+	rsp, err := k.ApplyMessage(msg, nil, true)
+	suite.Require().NoError(err)
+	suite.Require().False(rsp.Failed())
+	return crypto.CreateAddress(suite.from, nonce)
+}
+
+func (suite *EvmTestSuite) TestGasWhenReverted() {
```

Comment:
can we refactor this test? it's not maintainable

Reply:
I realize we don't need a special contract to reproduce the contract revert, so I changed it to the existing erc20 contract, is it good now?